### DISCUSSION TITLE
TCP defintion for UR different to teach panel?

### DISCUSSION THREAD

**Discussion by KonradJuenger** [visose/Robots#257] [Q&A] [answered]
Hi,

when I measure the tcp with the 4 Points method with the UR teach tablet, and I put the values in Robots/tool definiton the resulting UR script file differs from the intended TCP; 

_Is this due to a mistake in the Grasshopper file or a bug in Robots?_

#### On the robot: 

![nomacs_qwJdFp8G7t](https://github.com/visose/Robots/assets/76114767/c7ddc862-de48-4319-b274-db19c5956c8b)

![nomacs_oTk5mCOvxT](https://github.com/visose/Robots/assets/76114767/9d4fd805-fa02-4e48-b685-557a133b8fd8)

#### In Grasshopper:
![Rhino_mEe4xVL8S7](https://github.com/visose/Robots/assets/76114767/e22d8b39-6bc1-45f8-8a62-58ea3d544374)

#### My current workaround constists of this: 
![image](https://github.com/visose/Robots/assets/76114767/4083ec8e-2370-486a-bb5d-8b720658f658)
[UR_tcp.zip](https://github.com/visose/Robots/files/13862123/UR_tcp.zip)

**Comment by visose** [comment, marked as answer]
The X and Y switch is intended. The robots plugin always uses the same convention no matter the robot model. The output program will have a conversion from the Robots convention to the convention of the robot model.

**Reply by visose** [reply]
use the GetPlane/FromPlane components to convert the values in the pendant to the Robots convention.
